docs/api/lib/index.js

```diff
@@ -0,0 +1,153 @@
+const _ = require('lodash')
+const joiToSwagger = require('joi-to-swagger')
+
+const errorDocsLib = require('docs/api/lib/error')
+const extendedDocs = require('docs/api/lib/extended')
+
+/**
+ * Get methods for swagger routes
+ */
+const documentedMethods = ['GET', 'PUT', 'PATCH', 'DELETE', 'POST']
+const getMethod = methods => _.find(methods, method => _.includes(documentedMethods, method))
+
+/**
+ * Format parameters in query, params, body, header for docs
+ */
+const getFormattedValidationParams = (docsInKey, requiredKeys) => (obj, key) => {
+  const { description, ...schema } = obj
+  const isRequired = _.includes(requiredKeys, key)
+
+  return {
+    description,
+    in: docsInKey,
+    name: key,
+    required: isRequired,
+    schema,
+  }
+}
+
+function validateSection (objSchema, section) {
+  if (_.isEmpty(objSchema)) {
+    return []
+  }
+
+  const { swagger } = joiToSwagger(objSchema)
+  return _.map(swagger.properties, getFormattedValidationParams(section, swagger.required))
+}
+
+/**
+ * Get swagger validation objects for query, params, body, header
+ */
+function getValidationSections (routeStack) {
+  const validationMiddleware = {
+    auth: [],
+  }
+
+  _.each(routeStack, middleware => {
+    if (middleware.name === '_validator') {
+      middleware(validationMiddleware)
+    }
+
+    if (middleware.name === '_errorValidation') {
+      middleware(validationMiddleware)
+    }
+
+    // TODO can be expanded for other auth
+    if (middleware.name === 'jwt') {
+      validationMiddleware.auth.push(middleware.name)
+    }
+  })
```

Comment:
Instead of relaying on function names (and to assume no other middlewares will have same names), I think, it's more robust to `_.set(_validate, ['docValidation', target], sheme)` or `jwt.docAuth = {type: 'jwt'}`.

```suggestion
  const routeValidations = {}
  const routeAuths = []
  
  for (const middleware of routeStack) {
    if (middleware.docAuth) {
      routeAuths.push(middleware.docAuth)
    }
    if (middleware.docValidation) {
      Object.assign(routeValidations, middleware.docValidation)
    }
  }
```